code/modules/library/library_equipment.dm

```diff
@@ -0,0 +1,418 @@
+	b.title = selected_content.title ? selected_content.title : "no title" //The less of these that are null the better
+	b.author = selected_content.author ? selected_content.author : "no author"
+	b.summary = selected_content.summary ? selected_content.summary : "No Summary Provided"
+	b.categories = selected_content.categories ? selected_content.categories : list()
+	b.icon_state = "book[rand(1,16)]"
+	printing = FALSE
+
+/*
+ * Barcode Scanner
+ */
+/obj/item/barcodescanner
+	name = "barcode scanner"
+	desc = "A scanner used for managing library books, one can connect it the library system by tapping on a computer with it in hand."
+	icon = 'icons/obj/library.dmi'
+	icon_state ="scanner"
+	throw_speed = 1
+	throw_range = 5
+	w_class = WEIGHT_CLASS_TINY
+	var/list/modes = list(BARCODE_MODE_SCAN_SELECT, BARCODE_MODE_SCAN_INVENTORY, BARCODE_MODE_CHECKOUT, BARCODE_MODE_CHECKIN)
+	/// Associated Library Computer, needed to perform actions
+	var/obj/machinery/computer/library/computer
+	var/mode = BARCODE_MODE_SCAN_SELECT
+
+/obj/item/barcodescanner/attack_self(mob/user as mob)
+	if(!check_connection(user))
+		return
+	mode++
+	if(mode > length(modes))
+		mode = modes[1]
+	var/modedesc
+	switch(mode)
+		if(BARCODE_MODE_SCAN_SELECT)
+			modedesc = "Scan book to computer."
+		if(BARCODE_MODE_SCAN_INVENTORY)
+			modedesc = "Scan book into to general inventory."
+		if(BARCODE_MODE_CHECKOUT)
+			modedesc = "Checkout Book"
+		if(BARCODE_MODE_CHECKIN)
+			modedesc = "Checkin Book"
+		else
+			modedesc = "ERROR"
+	playsound(src, 'sound/machines/terminal_select.ogg', 15, TRUE)
+	to_chat(user, "[src] mode: [modedesc]")
+
+/obj/item/barcodescanner/proc/scanID(obj/item/card/id/ID, mob/user as mob)
+	if(!check_connection(user))
+		return
+
+	if(ID.registered_name)
+		computer.user_data.patron_name = ID.registered_name
+	else
+		computer.user_data.patron_name = null
+		computer.user_data.patron_account = null //account number should reset every scan so we don't accidently have an account number but no name
+		playsound(src, 'sound/machines/synth_no.ogg', 15, TRUE)
+		to_chat(user, "[src]'s screen flashes: 'ERROR! No name associated with this ID Card'")
+		return //no point in continuing if the ID card has no associated name!
+
+	playsound(src, 'sound/items/scannerbeep.ogg', 15, TRUE)
+	if(ID.associated_account_number)
+		computer.user_data.patron_account = ID.associated_account_number
+	else
+		computer.user_data.patron_account = null
+		to_chat(user, "[src]'s screen flashes: 'WARNING! Patron without associated account number Selected'")
+		return
+
+	to_chat(user, "[src]'s screen flashes: 'Patron Selected'")
+
+/obj/item/barcodescanner/proc/scanBook(obj/item/book/B, mob/user as mob)
+	if(!check_connection(user))
+		return
+
+	switch(mode)
+		if(BARCODE_MODE_SCAN_SELECT)
+			computer.select_book(B)
+			playsound(src, 'sound/machines/terminal_select.ogg', 15, TRUE)
+			to_chat(user, "[src]'s screen flashes: 'Book selected in library computer.'")
```

Comment:
All the `to_chat`s in this proc need spans.

Reply:
added to_chats